### PR TITLE
Support simple file stored repository

### PR DESCRIPTION
**Pull request by qrtt1**
Main idea:

1. build the first `real` repository for proof of concept
2. minor fixes `game-service` for unsaved state

### CODE REVIEW

File: love_letter/repository/__init__.py [hunk 1]
```diff
@@ -1,9 +1,17 @@
 import abc
+import logging
+import os.path
```
Comment (iannbing) [inline love_letter/repository/__init__.py:3]: Here `os.path` is imported, but in the code, `os` is used (`os.path` and `os.environ`). I guess you probably only want to import `os`?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -22,10 +22,9 @@ jobs:
...
-          poetry update
+          ./tests/run-tests.sh
 
-          export PYTHONPATH=$(pwd)
-          poetry run py.test --cov=love_letter --cov-report term tests
-          poetry run behave
+      - name: Run unit tests with real storage
+        run: |
+          export repository_impl=pickle
+          ./tests/run-tests.sh
```
Comment (iannbing) [inline .github/workflows/tests.yml:30]: Perhaps leave an empty line at the end of the file?

File: love_letter/models/cards.py [hunk 3]
```diff
@@ -87,6 +87,8 @@ def usage(self):
         }
 
     def __eq__(self, other):
```
Comment (iannbing) [inline love_letter/models/cards.py:89]: [nit] Interesting that this change allows `card == "prince"`. It feels somewhat awkward (wouldn't `card.name == "prince"` more expressive?)

File: love_letter/repository/__init__.py (hunk 1 above)
Comment (iannbing) [reply love_letter/repository/__init__.py:3]: Yet I'm surprised that the tests pass. Was there something false-positive?

File: tests/run-tests.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+pip install --upgrade pip
+pip install poetry
+poetry update
+
+export PYTHONPATH=$(pwd)
+poetry run py.test -v -s --cov=love_letter --cov-report term tests
+poetry run behave
```
Comment (iannbing) [inline tests/run-tests.sh:9]: Same here, maybe leave an empty line at the end of the file as well?

File: tests/test_card_behave.py [hunk 5]
```diff
@@ -5,24 +5,31 @@
 from love_letter.web.dto import GuessCard, ToSomeoneCard
 
 
+class _TestDeckForCardBehave(Deck):
```
Comment (iannbing) [inline tests/test_card_behave.py:8]: Nice clean-up! 👍  I thought the original implementation was a common practice in testing in Python 😌

File: love_letter/models/cards.py (hunk 3 above)
Comment (iannbing) [reply love_letter/models/cards.py:89]: By the way, was this change used anywhere in our codebase?

**Review by iannbing — COMMENTED** [review]
Awesome work! Just left some questions and some nitpicks. Curious to know what you think 😄

File: love_letter/repository/__init__.py (hunk 1 above)
Comment (qrtt1) [reply love_letter/repository/__init__.py:4]: I am not sure, but it works :P 

Anywhere, I added it.

File: love_letter/models/cards.py (hunk 3 above)
Comment (qrtt1) [reply love_letter/models/cards.py]: Revert this changes. We could pass all tests in both configurations (in-memory or file-store)

**Review by iannbing — APPROVED** [review]

File: love_letter/repository/__init__.py [hunk 6]
```diff
@@ -16,6 +24,29 @@ def get(self, game_id) -> Game:
         pass
 
 
+class GameRepositoryPickleImpl(GameRepository):
+    def __init__(self):
+        self.working_dir = tempfile.mkdtemp()
+        logger.info(
+            f"{GameRepositoryPickleImpl.__name__} working_dir: {self.working_dir}"
+        )
```
Comment (s9891326) [inline love_letter/repository/__init__.py:33]: 小建議，Python 3.10中 f-string可以改成這樣的寫法

```python
logger.info(f"{GameRepositoryPickleImpl.__name__} {self.working_dir=}")
```

Comment (qrtt1) [reply love_letter/repository/__init__.py]: 學習惹！

**Pull request merged**